### PR TITLE
feature: make overlay responsive

### PR DESCRIPTION
**Pull request by ArondenOuden**
Expanding the 'near you overlay' to have a desktop version and switch to the appropriate version depending on your screen size

### CODE REVIEW

File: static/css/home.css [hunk 1]
```diff
@@ -326,18 +326,95 @@ body{
     flex-direction: column;
     align-items: center;
     justify-content: flex-start;
-     /* Enable scrolling if content exceeds screen height */
     border-radius: 30px 30px 0px 0px;
     box-shadow: 0px -3px 5px rgba(0, 0, 0, 0.5);
 }
 
-#nearYouMobileLine {
+#nearYouMobileOverlay.desktop {
```
Comment (SilasPeters) [inline static/css/home.css]: A mobile overlay for desktop? It's a confusing name

File: static/js/common.js [hunk 2]
```diff
@@ -15,10 +15,17 @@ const emailInput = document.getElementsByClassName('successEmail')[0];
...
 var imageFile;
 var selectedLogo;
 var selectedLogoId;
 var emailCode;
 
 var pointersOnMap = [];
 var logoIcons = [];
+
+// Define the function before the event listener
+function isMobile() {
```
Comment (SilasPeters) [inline static/js/common.js:29]: There are more official methods to determine if you are on mobile. However, imagine a flip phone held horizontally: we would like to treat it as a desktop I guess. So this is fine

File: static/js/nearYouOverlay.js [hunk 3]
```diff
@@ -33,9 +41,120 @@ var Overlay = L.Class.extend({
...
+
+        // This styling needs to be in the javascript because it needs to override the onTouchMove function
+        this._overlayElement.style.bottom = "-90%";
+
+        this._overlayElement.scrollTo(0, 0);
+        let stickerDivs = document.querySelectorAll('.stickerDiv');
+        stickerDivs.forEach(div => div.classList.remove('revealed'));
+    },
+
+    openDesktopSidebar: function () {
```
Comment (SilasPeters) [inline static/js/nearYouOverlay.js:141]: You have quite some repetitive code for the mobile and desktop open and close and such. Perhaps it is better to abstract the real logic in a method, and give parameters such as `mobile, open`.

**Review by SilasPeters — CHANGES_REQUESTED** [review]
Seems adaptable and scalable! Although the code is quite repetitive, which makes it harder to maintain. Always try to define logic once, in one method, and give it parameters to make it work for different contexts.

File: static/css/home.css (hunk 1 above)
Comment (ArondenOuden) [reply static/css/home.css]: I renamed the overlay and the top text.
The line is only for mobile, so it still has 'mobile' in the name.
The button is only for desktop, so it still has 'desktop' in the name.

File: static/js/common.js (hunk 2 above)
Comment (ArondenOuden) [reply static/js/common.js:29]: There are indeed methods to [detect a mobile browser](https://stackoverflow.com/questions/11381673/detecting-a-mobile-browser), but that doesn't really provide anything practical. As you said, a phone held horizontally would be better to treat as desktop. So therefore I will keep my current implementation.

File: static/js/nearYouOverlay.js (hunk 3 above)
Comment (ArondenOuden) [reply static/js/nearYouOverlay.js:141]: I rewrote the toggling of the overlay to be more readable and clean. I don't think there is much more to optimize in terms of repetitive code, so I'll resolve this conversation

**Review by SamStaijen — APPROVED** [review]
LGTM

**Review by SilasPeters — APPROVED** [review]

**Pull request merged**